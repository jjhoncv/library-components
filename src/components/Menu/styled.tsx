import styled from 'styled-components';
import { COLORS, FONTS } from './../../ui';

export const Menu: any = styled.ul`
  display: inline-block;
  background: ${COLORS.WHITE};
  box-shadow: 0 1px 4px rgba(0,0,0,0.25);
  border-radius: 2px;
  font-size: 14px;
  font-family: ${FONTS.ROBOTO_REGULAR};
  color: ${COLORS.TEXT_GRAY};
`;

export const ItemStyled = styled.li`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  padding: 4px 8px 4px 8px;
  box-sizing: border-box;
  cursor: pointer;

  svg {
    width: 15px;
    height: 15px;
    margin-left: 17px;

    & path {
      color: ${COLORS.TEXT_GRAY};
    }

    & polygon {
      color: ${COLORS.TEXT_GRAY};
    }
  }

  &:hover {
    background: ${COLORS.HOVER_GRAY};
  }

  &:first-child {
    font-family: ${FONTS.ROBOTO_BOLD};
  }
`;
