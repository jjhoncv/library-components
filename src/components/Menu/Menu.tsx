import * as React from 'react';
import * as PropTypes from 'prop-types';
import { Menu, ItemStyled } from './styled';
import * as Icons from 'react-feather';

interface Props {
  children?: string;
  icon?: string;
  onClick?: Function;
}

const Item: React.FC<Props> = ({
  children,
  icon,
  onClick
}) => {
  const Icon = ({ icon }) => {
    const nameIconCapitalize = (icon.charAt(0).toUpperCase() + icon.slice(1)).toString();
    const IconName = Icons[nameIconCapitalize];
    return <IconName />
  }

  return (
    <ItemStyled onClick={onClick}>
      { children }
      { icon &&
        <Icon icon={icon} /> }
    </ItemStyled>
  );
};

Item.propTypes = {
  children: PropTypes.string,
  icon: PropTypes.string,
  onClick: PropTypes.func,
}

Menu.Item = Item;

export { Menu };
