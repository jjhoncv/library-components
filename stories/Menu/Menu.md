# Menu

Importaci√≥n del componente Menu.

```js
import { Menu } from 'library/components/Menu';
```

<!-- STORY -->

#### Uso

```jsx
<Menu>
  <Menu.Item>
    JOHN
  </Menu.Item>
  <Menu.Item icon="edit" onClick={()=> console.log('Perfil')}>
    Perfil
  </Menu.Item>
  <Menu.Item onClick={()=> console.log('Salir')}>
    Salir
  </Menu.Item>
</Menu>
```
