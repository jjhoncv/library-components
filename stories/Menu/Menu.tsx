import * as React from 'react';
import { WithPropTypes } from '../../.storybook/Containers/WithPropTypes';
import { Menu } from '../../src/components/Menu';

const Wrapper = () => (
  <Menu>
    <Menu.Item>
      JOHN
    </Menu.Item>
    <Menu.Item icon="edit" onClick={()=> console.log('Perfil')}>
      Perfil
    </Menu.Item>
    <Menu.Item onClick={()=> console.log('Salir')}>
      Salir
    </Menu.Item>
  </Menu>
);

export default () => {
  return WithPropTypes({
    Component: Menu.Item,
    RenderComponent: Wrapper,
    showProp: false
  });
}
