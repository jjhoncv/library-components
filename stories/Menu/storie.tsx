import { storiesOf } from '@storybook/react';
import Menu from './Menu';
import * as readme from './Menu.md';

const stories = storiesOf('2. Components/Menu', module);

stories.add('Default', Menu(), {
  readme: { content: readme }
})
