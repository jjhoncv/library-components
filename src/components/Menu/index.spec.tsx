import * as React from 'react';
import { shallow } from 'enzyme';
import { Menu } from './Menu';

<Menu>
  <Menu.Item>
    JOHN
  </Menu.Item>
</Menu>

describe('Menu Component', () => {
  describe('Menu => render snapshot', () => {
    it('Render correctly', () => {
      const wrapper = shallow(
        <Menu>
          <Menu.Item>
            JOHN
          </Menu.Item>
        </Menu>
      );
      expect(wrapper.html()).toMatchSnapshot();
    })
  });

  describe('Menu Builder', () => {
    it('Menu => Render children', () => {
      const wrapper = shallow(
        <Menu>
          <Menu.Item>
            USER
          </Menu.Item>
        </Menu>
      );
      expect(wrapper.find(Menu.Item).prop('children')).toBe('USER');
    });

    it('Menu => Simulate click', () => {
      const onClickMock = jest.fn();
      const wrapper = shallow(
        <Menu>
          <Menu.Item onClick={onClickMock}>
            USER
          </Menu.Item>
        </Menu>
      );
      wrapper.find(Menu.Item).simulate('click');
      expect(onClickMock).toHaveBeenCalled();
    });
  });
})
