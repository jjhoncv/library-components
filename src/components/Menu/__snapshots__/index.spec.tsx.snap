// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component Menu => render snapshot Render correctly 1`] = `"<ul class=\\"sc-bdVaJa kdFyRA\\"><li class=\\"sc-bwzfXH eVnbzd\\">JOHN</li></ul>"`;
